airflow/providers/databricks/hooks/databricks.py

```diff
@@ -151,6 +158,89 @@ def _parse_host(host: str) -> str:
             # In this case, host = xx.cloud.databricks.com
             return host
 
+    def _get_aad_token(self, resource: str) -> str:
+        """
+        Function to get AAD token for given resource. Supports managed identity or service principal auth
+        :param resource: resource to issue token to
+        :return: AAD token, or raise an exception
+        """
+        if resource in self.aad_tokens:
+            d = self.aad_tokens[resource]
+            now = int(time.time())
+            if d['expires_on'] > (now - TOKEN_REFRESH_LEAD_TIME):  # it expires in more than 5 minutes
+                return d['token']
+            self.log.info("Existing AAD token is expired, or going to expire soon. Refreshing...")
+
+        attempt_num = 1
+        while True:
+            try:
+                if self.databricks_conn.extra_dejson.get('use_azure_managed_identity', False):
+                    # https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token#get-a-token-using-http
+                    params = {
+                        "api-version": "2018-02-01",
+                        "resource": resource,
+                    }
+                    resp = requests.get(
+                        "http://169.254.169.254/metadata/identity/oauth2/token",
```

Comment:
Why fixed IP address here? This sounds very wrong

Reply:
I understanda this is azrure's metadata server? But is there no better way to reach the metadata server (and should we only limit it if we check in the environment that we are running on Azure managed vm ? 

For example in Google's VM you can use `metadata.google.internal` name https://cloud.google.com/compute/docs/metadata/overview

Reply:
it's a fixed address for internal metadata service - see the linked documentation

Reply:
Yeah - but see my comment above. At the very least (if we cannot get symbolic name) this address should be extracte and commented out as "AZURE_METADATA_IP" for example and explained, without the need of reading the docs actually.

Reply:
And we should likely check if we are in Azure VM (and fail if we aren't) without even reaching out to the metadata server. Not really necessary but It would be nice to check it (via env vars I guess) before - otherwise you might get strange errors when you enable it by mistake on non-azure managed-identity server. Google's meta-data servers I think have the same IP address, so the responses from it might be confusing.

Reply:
See for example the comment here (accepted answer is to use metadata server but the comment is about long timeouts and confusing behaviour on non-azure instances): https://stackoverflow.com/questions/54913218/detect-whether-running-on-azure-iaas-vm-net-application